### DISCUSSION TITLE
Extract form-filling values from a PDF document that supports Acroform

### DISCUSSION THREAD

**Discussion by hoangthanh283** [pypdfium2-team/pypdfium2#322] [Q&A] [answered]
### Checklist

- [ ] I confirm this is not a question or feature request. Otherwise, use the Discussions page. *Note: Was edited out by reporter, added back by @mara004.*
- [X] I confirm this is **not an issue encountered with an installed build of pypdfium2, but about some other aspect of the project** (specify below). *Otherwise, use one of the package templates (PyPA/conda), even if you believe this is not a package-specific issue.*
- [X] I confirm this is **not about an unofficial build of pypdfium2**. *We do not support third-party builds, and they are not eligible for a bug report.*

### Reason for Generic issue (keyword/topic)

Looks like that pdfium supports to extract values in a form-fill PDF file but there is no visible API to do that

### Description

I can check that in this document, the PDFMiner supports to detect if the document contains form-fill values or not. But there is no clearly API from the official document to do the form-fill value extraction.
For example, in this document (attached as below):

```Python
pdf = PdfDocument(pdf_path)
print(pdf.get_formtype()) #  = 1
```
Can you please show me what API should I use to extract the form-fill values.
[MedicalAttorney-6.pdf](https://github.com/user-attachments/files/16246340/MedicalAttorney-6.pdf)

**Comment by mara004** [comment]
This *is* a feature request, and you are not supposed to edit the template! [^1]
Converting to discussion. (source issue: #321)

[^1]: This is not the first time I've seen deliberate template edits here to bypass organization rules, and am starting to understand `yt-dlp`'s rigorous stance of blocking anyone who does...

**Comment by mara004** [comment, marked as answer]
As to the forms, no, pypdfium2 does not provide (or plan to provide) helpers for this.
pdfium is a vast library, and it is infeasible for a single volunteer to write wrappers around all of its APIs, so I am focusing on a few core support models.

However, sufficiently experienced embedders may access the [raw pdfium API](https://github.com/pypdfium2-team/pypdfium2?tab=readme-ov-file#raw-pdfium-api) directly if you need features beyond what is covered by the support models. In this case, one would refer to the APIs and docs in [`fpdf_formfill.h`](https://pdfium.googlesource.com/pdfium/+/refs/heads/main/public/fpdf_formfill.h).
